### PR TITLE
Add ElasticSearchDom W/Out EncrypAtRestOpt #1089

### LINKED ISSUE #1089: Add Elastic Search Domain Without EncryptionAtRestOptions query for AWS CloudFormation
### Platform
CloudFormation

### Provider
AWS

### Description
Elastic Search Domain should have encrypt data at rest.

### PR DESCRIPTION
**Pull request by joelcarvalhocheckmarx**
Elastic Search Domain should have encrypt data at rest.

Closes #1089

### CODE REVIEW

File: assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "elastic_search_domain_without_encryption_at_rest_options",
+  "queryName": "Elastic Search Domain Without EncryptionAtRestOptions",
+  "severity": "MEDIUM",
+  "category": null,
+  "descriptionText": "Elastic Search Domain should have encrypt data at rest.",
+  "descriptionUrl": "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticsearch-domain.html#cfn-elasticsearch-domain-encryptionatrestoptions"
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/metadata.json:8]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/test/negative.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        Version: "2012-10-17"
+        Statement:
+          -
+            Effect: "Allow"
+            Principal:
+              AWS: "arn:aws:iam::123456789012:user/es-user"
+            Action: "es:*"
+            Resource: "arn:aws:es:us-east-1:846973539254:domain/test/*"
+      AdvancedOptions:
+        rest.action.multi.allow_explicit_index: "true"
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/test/negative.yaml:34]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/test/positive.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        Version: "2012-10-17"
+        Statement:
+          -
+            Effect: "Allow"
+            Principal:
+              AWS: "arn:aws:iam::123456789012:user/es-user"
+            Action: "es:*"
+            Resource: "arn:aws:es:us-east-1:846973539254:domain/test/*"
+      AdvancedOptions:
+        rest.action.multi.allow_explicit_index: "true"
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/test/positive.yaml:69]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/query.rego [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package Cx
+
+CxPolicy [ result ] {
+  document := input.document
+  resource = document[i].Resources[name]
+  resource.Type == "AWS::Elasticsearch::Domain"
+  properties := resource.Properties
+  
+  exists_encryptionAtRestOptions := object.get(properties, "EncryptionAtRestOptions", "undefined") != "undefined"
+  not exists_encryptionAtRestOptions
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/query.rego:10]: Simplify lines 9, 10 to just:

`object.get(properties, "EncryptionAtRestOptions", "undefined") == "undefined"`

File: assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/query.rego [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+              }
+}
+
+CxPolicy [ result ] {
+  document := input.document
+  resource = document[i].Resources[name]
+  resource.Type == "AWS::Elasticsearch::Domain"
+  properties := resource.Properties
+  
+  checks_encryption_at_rest_options(properties.EncryptionAtRestOptions)
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/query.rego:27]: This function is only being used once, so it's unnecessary, this should be enough:

`properties.EncryptionAtRestOptions.Enabled  != true`

File: assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/metadata.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "elastic_search_domain_without_encryption_at_rest_options",
+  "queryName": "Elastic Search Domain Without EncryptionAtRestOptions",
+  "severity": "MEDIUM",
+  "category": null,
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/metadata.json:5]: `Encryption and Key Management`

File: assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/metadata.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "elastic_search_domain_without_encryption_at_rest_options",
+  "queryName": "Elastic Search Domain Without EncryptionAtRestOptions",
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/elastic_search_domain_without_encryption_at_rest_options/metadata.json:3]: **Directory name**: `elastic_search_domain_encryption_at_rest_disabled`

```
"id": "elastic_search_domain_encryption_at_rest_disabled",
"queryName": "Elastic Search Domain Encryption At Rest Disabled",
```

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Requesting changes

**Comment by rogeriopeixotocx** [comment]
LGTM ✅

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**